packages/devtools_server/lib/src/server.dart

```diff
@@ -281,6 +282,57 @@ Future<void> _handleVmRegister(
   }
 }
 
+Future<void> _handleDevToolsLaunch(
+  dynamic id,
+  Map<String, dynamic> params,
+  bool machineMode,
+  bool headlessMode,
+  String devToolsUrl,
+) async {
+  if (!params.containsKey('vmServiceUri')) {
+    printOutput(
+      'Invalid input: $params does not contain the key \'vmServiceUri\'',
+      {
+        'id': id,
+        'error':
+            'Invalid input: $params does not contain the key \'vmServiceUri\'',
+      },
+      machineMode: machineMode,
+    );
+  }
+
+  // params['vmServiceUri'] should contain a vm service uri.
+  final vmServiceUri = Uri.tryParse(params['vmServiceUri']);
```

Comment:
Should this be `uri`? We should add this param to the daemon.md doc.

Reply:
I called this `vmServiceUri` to avoid ambiguity (since DevTools has a URI that it's hosted on... for `vm.register` it seemed clearer a param named `uri` would be for the VM service, but for "launch DevTools" that seemed less so?). (04f1ac54)

Reply:
👍